mlir/test/Conversion/GPUCommon/lower-memcpy-to-gpu-runtime-calls.mlir

```diff
@@ -17,3 +17,23 @@ module attributes {gpu.container_module} {
     return
   }
 }
+
+// -----
+
+module attributes {gpu.container_module} {
+
+  // CHECK: func @dynamic
+  func.func @dynamic(%dst : memref<?x?xf32, 1>, %src : memref<?x?xf32>) {
+    // CHECK: %[[T0:.*]] = llvm.call @mgpuStreamCreate
+    %t0 = gpu.wait async
+    %t1 = gpu.memcpy async [%t0] %dst, %src : memref<?x?xf32, 1>, memref<?x?xf32>
```

Comment:
Can you also improve the verifier to reject memrefs that do not have identity layout?

Reply:
Done

Reply:
Actually enforcing this check in verifier is causing problems with [this](https://github.com/llvm/llvm-project/blob/main/mlir/test/Dialect/SparseTensor/GPU/gpu_matvec_lib.mlir#L24) test

Reply:
Is that the only failing test? Maybe we can put a `memref.cast` in the sparse compiler to cast away the layout map, along with a TODO for the sparse compiler folks to change the way `CoordinatesOp` is generated.

I'm also wondering if the IR generated by the sparse compiler here is actually correct. As in: does the memcpy to GPU actually work, given that there is a layout map? @aartbik @PeimingLiu

Reply:
Before this PR, the implementation calculated the number of elements as `stride[0] * size[0]`. Not sure why it was like that. But maybe the sparse test will start failing now that we don't take into account the strides anymore.

Reply:
I am not adding the verifier for gpu.memcpy as a part of this PR as the sparse case is failing and can be handled separately. The bug fix for computing number of elements can go in as part of this PR